src/EditorFeatures/CSharpTest/RawStringLiteral/RawStringLiteralCommandHandlerTests.cs

```diff
@@ -319,6 +319,72 @@ public void TestReturnInEmptyFile()
         testState.SendReturn(handled: false);
     }
 
+    [WpfFact]
+    public void TestReturnAfterThreeQuotesFollowingText()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = """$$following text""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = """
+                $$following text
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnAfterThreeQuotesFollowingText_Interpolated()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $"""$$following text {0}""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = $"""
+                $$following text {0}
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnAfterTextInRawStringFollowingText()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = """before text$$following text""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = """
+                before text
+                $$following text
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnAfterTextInRawStringFollowingText_Interpolated()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $"""before text$$following text {0}""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = $"""
+                before text
+                $$following text {0}
+                """;
+            """");
+    }
```

Comment:
Add tests inside an interpolation. 
Add tests before a quote that is or is not part of the end delimiter.

Reply:
Add tests when caret is on the {

Reply:
Added more tests and supported the above cases ptal